### PR TITLE
build: Cleanup Uno.CrossTargetting imports

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by Youssef1313** [comment]
@jeromelaban Can you take a look?

**Comment by Youssef1313** [comment]
@jeromelaban Can you take a look please?

File: src/SourceGenerators/XamlGenerationTests/GlobalStaticResources02.xaml.cs [hunk 1]
```diff
@@ -1,17 +0,0 @@
...
-using System;
-using System.Collections.Generic;
-using System.Text;
-using XamlGenerationTests;
-
-namespace XamlGenerationTests
-{
-	public sealed partial class GlobalStaticResources
-	{
-		public GlobalStaticResources()
```
[the remark this answers is not in the document]
Comment (Youssef1313) [reply src/SourceGenerators/XamlGenerationTests/GlobalStaticResources02.xaml.cs:11]: @jeromelaban The file is enclosed with `#if XAMARIN`, but previously `XAMARIN` never existed in DefineConstants. Now `XAMARIN` comes properly from `Uno.CrossTargetting` (`XamlGenerationTests.csproj` didn't use to include `Uno.CrossTargetting`)

And the reality is that `ColorResource_asName` doesn't actually exist. I guess that is an existing bug.

Maybe I should add the file back replacing `#if XAMARIN` with `#if false // add link to a new issue`?

File: src/Uno.UWP/Graphics/Imaging/BitmapEncoder.cs [hunk 2]
```diff
@@ -1,26 +1,78 @@
...
+	private static readonly IDictionary<Guid, SKEncodedImageFormat> _encoderMap =
+		new Dictionary<Guid, SKEncodedImageFormat>()
+		{
+			{BmpEncoderId, SKEncodedImageFormat.Bmp},
+			{GifEncoderId, SKEncodedImageFormat.Gif},
+			{JpegEncoderId, SKEncodedImageFormat.Jpeg},
+			{PngEncoderId, SKEncodedImageFormat.Png},
+			{HeifEncoderId, SKEncodedImageFormat.Heif},
+		};
+#endif
```
[the remark this answers is not in the document]
Comment (Youssef1313) [reply src/Uno.UWP/Graphics/Imaging/BitmapEncoder.cs:77]: @jeromelaban Field initialization order is not deterministic across partial classes. Here we wanted to make sure these dictionaries are initialized after the giuds (ie, `JpegEncoderId`, etc..)

File: src/PlatformItemGroups.props [hunk 3]
```diff
@@ -15,85 +15,40 @@
 
 		<_IsNetStd>false</_IsNetStd>
 		<_IsNetStd Condition="'$(TargetFramework)'=='netstandard2.0' or $(_IsNetCore)">true</_IsNetStd>
-
```
[the remark this answers is not in the document]
Comment (Youssef1313) [reply src/PlatformItemGroups.props:18]: @jeromelaban I assumed that as long as the build passed then we are not defining something too late. The problem with evaluating these properties early is that `TargetFramework` wouldn't have been set.

[the remark this answers is not in the document]
Comment (Youssef1313) [reply src/PlatformItemGroups.props:18]: Note: all references to these properties are boolean references, ie, we use the condition as just `$(IsCatalyst)` not as `'$(IsCatalyst)'=='true'`, which means that if `IsCatalyst` is not yet defined MSBuild will complain and fail the build.

[the remark this answers is not in the document]
Comment (Youssef1313) [reply src/PlatformItemGroups.props:18]: I can rename this from `props` to `targets` for clarity though as I agree `props` is confusing

File: src/Directory.Build.targets [hunk 4]
```diff
@@ -1,9 +1,11 @@
 <Project ToolsVersion="15.0">
   <Import Project="TargetFrameworkSelection.targets" />
+  <Import Project="uno.mobile-cross-targeting-disable.props" />
+  <Import Project="Uno.CrossTargetting.targets" />
```
[the remark this answers is not in the document]
Comment (Youssef1313) [reply src/Directory.Build.targets:4]: All properties are conditioned on `TargetFramework`, so I think we have to be in targets for `TargetFramework` to be already defined.

File: src/Uno.UI/Uno.UI.csproj [hunk 5]
```diff
@@ -25,9 +25,6 @@
 
 		<UnoRuntimeIdentifier Condition="'$(TargetFramework)'=='netstandard2.0'">Reference</UnoRuntimeIdentifier>
 
-		<!-- Ensures the .xr.xml files are generated in a proper layout folder -->
-		<GenerateLibraryLayout>true</GenerateLibraryLayout>
```
[the remark this answers is not in the document]
Comment (Youssef1313) [reply src/Uno.UI/Uno.UI.csproj:29]: @jeromelaban It is already in Uno.CrossTargetting

File: src/SourceGenerators/XamlGenerationTests/GlobalStaticResourcesTest.cs [hunk 6]
```diff
@@ -1,20 +0,0 @@
...
-using System.Text;
-using XamlGenerationTests;
-
-namespace XamlGenerationTests.Shared
-{
-	public static class GlobalStaticResourcesTest
-	{
-		static GlobalStaticResourcesTest()
-		{
-			var res = GlobalStaticResources.OtherResources01.ToString();
```
[the remark this answers is not in the document]
Comment (Youssef1313) [reply src/SourceGenerators/XamlGenerationTests/GlobalStaticResourcesTest.cs:13]: Same as above.

File: src/Directory.Build.targets (hunk 4 above)
[the remark this answers is not in the document]
Comment (Youssef1313) [reply src/Directory.Build.targets]: This has to be in targets to ensure we got `UnoRuntimeIdentifier` set properly.

**Review by jeromelaban — APPROVED** [review]

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**